metapool/sample_sheet.py

```diff
@@ -803,6 +816,119 @@ def __init__(self, path=None):
         }
 
 
+class MetagenomicSampleSheetv101(KLSampleSheet):
+    # Adds support for optional KATHAROSEQ columns in [Data] section.
+
+    _HEADER = {
+        'IEMFileVersion': '4',
+        'SheetType': _STANDARD_METAG_SHEET_TYPE,
+        'SheetVersion': '101',
+        'Investigator Name': 'Knight',
+        'Experiment Name': 'RKL_experiment',
+        'Date': None,
+        'Workflow': 'GenerateFASTQ',
+        'Application': 'FASTQ Only',
+        'Assay': _METAGENOMIC,
+        'Description': '',
+        'Chemistry': 'Default',
+    }
+
+    data_columns = ['Sample_ID', 'Sample_Name', 'Sample_Plate', 'well_id_384',
+                    'I7_Index_ID', 'index', 'I5_Index_ID', 'index2',
+                    'Sample_Project', 'Well_description']
+
+    # Names of columns needed in the sample-information-file output that is
+    # later passed into Qiita:
+    # 'katharoseq_aliquot_volume_ul',
+    # 'katharoseq_batch_information',
+    # 'katharoseq_cell_count',
+    # 'katharoseq_plate_number',
+    # 'katharoseq_strain'
+
+    # columns present in an pre-prep file (amplicon) that included katharoseq
+    # controls. Presumably we will need these same columns in a sample-sheet.
+    optional_katharoseq_columns = ['Kathseq_RackID', 'TubeCode',
+                                   'katharo_description',
+                                   'number_of_cells',
+                                   'platemap_generation_date',
+                                   'project_abbreviation',
+                                   'vol_extracted_elution_ul', 'well_id_96']
+
+    # For now, assume only MetagenomicSampleSheetv101 (v100, v95, v99) contains
+    # 'contains_replicates' column. Assume AbsQuantSampleSheetv10 doesn't.
+    _BIOINFORMATICS_COLUMNS = {'Sample_Project', 'QiitaID', 'BarcodesAreRC',
+                               'ForwardAdapter', 'ReverseAdapter',
+                               'HumanFiltering', 'contains_replicates',
+                               'library_construction_protocol',
+                               'experiment_design_description'}
+
+    CARRIED_PREP_COLUMNS = ['experiment_design_description', 'i5_index_id',
+                            'i7_index_id', 'index', 'index2',
+                            'library_construction_protocol', 'sample_name',
+                            'sample_plate', 'sample_project',
+                            'well_description', 'well_id_384']
+
+    def __init__(self, path=None):
+        super().__init__(path=path)
+        self.remapper = {
+            # TODO: It seems like the optional columns need to be defined in
```

Comment:
Open issue and delete? If there are ToDo's please add them as PR, not as part of the code.

Reply:
ty for the catch! Issue was resolved.